src/state/trajectory/logics.ts

```diff
@@ -334,21 +346,93 @@ const loadFileViaUrl = createLogic({
 });
 
 const fileConversionLogic = createLogic({
-    process(deps: ReduxLogicDeps, dispatch) {
+    process(deps: ReduxLogicDeps, dispatch, done) {
         dispatch(
             setStatus({
                 status: VIEWER_IMPORTING,
             })
         );
+        done();
     },
     type: CONVERT_FILE,
 });
 
+const setConversionEngineLogic = createLogic({
+    async process(deps: ReduxLogicDeps): Promise<{
+        engineType: AvailableEngines;
+        template: Template;
+        templateMap: TemplateMap;
+    }> {
+        const {
+            httpClient,
+            action,
+            uiTemplateUrlRoot,
+            uiBaseTypes,
+            uiCustomTypes,
+            uiTemplateDownloadUrlRoot,
+        } = deps;
+        const baseTypes = await httpClient
+            .get(`${uiTemplateDownloadUrlRoot}/${uiBaseTypes}`)
+            .then((baseTypesReturn: AxiosResponse) => {
+                return baseTypesReturn.data;
+            });
+
+        const customTypes = await httpClient
+            .get(`${uiTemplateUrlRoot}/${uiCustomTypes}`)
+            .then((customTypesReturn: AxiosResponse) => {
+                return customTypesReturn.data;
+            })
+            .then((fileRefs) =>
+                Promise.all(
+                    map(
+                        fileRefs,
+                        async (ref) =>
+                            await httpClient
+                                .get(ref.download_url)
+                                .then((file) => file.data)
+                    )
+                )
+            );
+
+        const initTypeMap: TemplateMap = {};
+
+        const typeMap: TemplateMap = reduce(
+            customTypes,
```

Comment:
Might prefer native `map`/`reduce` here, unless there's a general preference in Simularium to use lodash. (The method syntax could also help to unwind the formatting of the `map` on 387 a bit)

Reply:
I think these are used here because the data coming in is actually an object instead of array

Reply:
ah, that makes sense - I don't use lodash enough to consistently remember that it lets you do that